WalletWasabi/Blockchain/Transactions/AllTransactionStore.cs

```diff
@@ -185,8 +196,9 @@ public IEnumerable<SmartTransaction> ReleaseToMempoolFromBlock(uint256 blockHash
 				if (ConfirmedStore.TryRemove(txHash, out var removedTx))
 				{
 					removedTx.SetUnconfirmed();
-					if (MempoolStore.TryAddOrUpdate(removedTx).isAdded)
+					if (MempoolStore.TryAdd(removedTx))
 					{
+						_ = MempoolStore.TryUpdate(removedTx);
```

Comment:
Isn't this change incorrect? If `TryAdd` returns `true`, then `TryUpdate` has no effect.
To match past behavior, and if I understand correctly, `_ = MempoolStore.TryUpdate(removedTx);` should be moved in an `else`